File: DuDoong-Api/src/main/java/band/gosrock/api/alimTalk/service/SendRegisterAlimTalkService.java
package band.gosrock.api.alimTalk.service;


import band.gosrock.domain.common.alarm.UserKakaoTalkAlarm;
import band.gosrock.infrastructure.config.alilmTalk.NcpHelper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class SendRegisterAlimTalkService {
    private final NcpHelper ncpHelper;

    public void execute(String userName, String to) {
        String content = UserKakaoTalkAlarm.creationOf(userName);
        ncpHelper.sendButtonNcpAlimTalk(to, UserKakaoTalkAlarm.creationTemplateCode(), content);
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/alimTalk/service/SendRegisterAlimTalkService.kt
package band.gosrock.api.alimTalk.service

import band.gosrock.domain.common.alarm.UserKakaoTalkAlarm
import band.gosrock.infrastructure.config.alilmTalk.NcpHelper
import org.springframework.stereotype.Service

@Service
class SendRegisterAlimTalkService(
    private val ncpHelper: NcpHelper,
) {
    fun execute(userName: String, to: String) {
        val content = UserKakaoTalkAlarm.creationOf(userName)
        ncpHelper.sendButtonNcpAlimTalk(to, UserKakaoTalkAlarm.creationTemplateCode(), content)
    }
}
